src/rewards/rewards.service.ts

```diff
@@ -41,4 +46,50 @@ export class RewardsService {
     }
     return claim;
   }
+
+  /**
+   * Get all token rewards emmited, by vault
+   * @param chain Network chain obj
+   * @param address Vault token adress
+   */
+  async rewardSchedulesByVault(chain: Chain, address: VaultDefinition['vaultToken']): Promise<RewardSchedulesByVault> {
+    const rewardsLogger = getRewordLoggerInst(chain);
+
+    if (!rewardsLogger) throw new NotFound(`No rewards token emission data for network ${Chain.name}`);
+
+    const vault = chain.vaults.find((v) => v.vaultToken === address);
+
+    if (!vault || !vault?.vaultToken) throw new NotFound(`Unknown vault ${address}`);
+
+    return {
+      schedules: await getRewardSchedules(chain, vault.vaultToken, rewardsLogger),
+    };
+  }
+
+  /**
+   * Get all token rewards emmited, by all vaults
+   */
+  async rewardSchedulesVaultsList(chain: Chain): Promise<RewardSchedulesByVaults> {
+    const rewardsLogger = getRewordLoggerInst(chain);
+
+    if (!rewardsLogger) throw new NotFound(`No rewards token emission data for network ${Chain.name}`);
+
+    const vaultsSchedules = await Promise.all(
+      chain.vaults.map(async (vault) => {
+        if (!vault.vaultToken) return [];
```

Comment:
i think this should never be undefined based on the typing as the configs would not compile